src/take.test.ts

```diff
@@ -16,3 +24,139 @@ describe("data_last", () => {
+    expectTypeOf(take(strict, 2)).toEqualTypeOf<[string, number]>();
+
+    const strictReadonly = ["1", 2, true] as readonly [string, number, boolean];
+    expectTypeOf(take(strictReadonly, 2)).toEqualTypeOf<[string, number]>();
+  });
+
+  it("handles literal unions", () => {
+    const unions = ["cat", "mouse"] as ["cat" | "dog", "mouse" | "chicken"];
+    expectTypeOf(take(unions, 1)).toEqualTypeOf<["cat" | "dog"]>();
+
+    const unionsReadonly = ["cat", "mouse"] as readonly [
+      "cat" | "dog",
+      "mouse" | "chicken",
+    ];
+    expectTypeOf(take(unionsReadonly, 1)).toEqualTypeOf<["cat" | "dog"]>();
+  });
+
+  it("handles rest tails", () => {
+    const restTail = ["1", 2, 3] as [string, ...Array<number>];
+    expectTypeOf(take(restTail, 1)).toEqualTypeOf<[string]>();
+    expectTypeOf(take(restTail, 2)).toEqualTypeOf<[string, ...Array<number>]>();
+
+    const restTailReadonly = ["1", 2, 3] as readonly [string, ...Array<number>];
+    expectTypeOf(take(restTailReadonly, 1)).toEqualTypeOf<[string]>();
+    expectTypeOf(take(restTailReadonly, 2)).toEqualTypeOf<
+      [string, ...Array<number>]
+    >();
+  });
+
+  it("handles rest heads", () => {
+    const restHead = [1, 2, 3, "foo"] as [...Array<number>, string];
+    expectTypeOf(take(restHead, 3)).toEqualTypeOf<[...Array<number>, string]>();
+
+    const restHeadReadonly = [1, 2, 3, "foo"] as readonly [
+      ...Array<number>,
+      string,
+    ];
+    expectTypeOf(take(restHeadReadonly, 3)).toEqualTypeOf<
+      [...Array<number>, string]
+    >();
+  });
+
+  it("handles a mix of unions and rest params", () => {
+    const mix = ["cat", 123, 123, 456] as ["cat" | "dog", ...Array<123 | 456>];
+    expectTypeOf(take(mix, 1)).toEqualTypeOf<["cat" | "dog"]>();
+    expectTypeOf(take(mix, 3)).toEqualTypeOf<
+      ["cat" | "dog", ...Array<123 | 456>]
+    >();
+
+    const mixReadonly = ["cat", 123, 123, 456] as readonly [
+      "cat" | "dog",
+      ...Array<123 | 456>,
+    ];
+    expectTypeOf(take(mixReadonly, 1)).toEqualTypeOf<["cat" | "dog"]>();
+    expectTypeOf(take(mixReadonly, 3)).toEqualTypeOf<
+      ["cat" | "dog", ...Array<123 | 456>]
+    >();
+  });
+
+  it("handles N greater than tuple length", () => {
+    const short = ["foo"] as [string];
+    expectTypeOf(take(short, 10)).toEqualTypeOf<[string]>();
+
+    const shortReadonly = ["foo"] as readonly [string];
+    expectTypeOf(take(shortReadonly, 10)).toEqualTypeOf<[string]>();
+  });
+
+  it("handles shorter heads", () => {
+    const shortHead = [1, "foo", true, false] as [
+      number,
+      string,
+      ...Array<boolean>,
+    ];
+    expectTypeOf(take(shortHead, 3)).toEqualTypeOf<
+      [number, string, ...Array<boolean>]
+    >();
+
+    const shortHeadReadonly = [1, "foo", true, false] as readonly [
+      number,
+      string,
+      ...Array<boolean>,
+    ];
+    expectTypeOf(take(shortHeadReadonly, 3)).toEqualTypeOf<
+      [number, string, ...Array<boolean>]
+    >();
+  });
```

Comment:
Is this test redundant, it feels like it's already tested as part of the tuples with rest tails...